open-metadata-implementation/admin-services/admin-services-server/src/main/java/org/odpi/openmetadata/adminservices/OMAGServerAdminForViewServices.java

```diff
@@ -313,11 +315,44 @@ public VoidResponse configureAllViewServices(String userId,
     private ViewServiceConfig createViewServiceConfig(ViewServiceRegistration     registration,
                                                       ViewServiceConfig           requestedViewServiceConfig)
     {
-        ViewServiceConfig viewServiceConfig = new ViewServiceConfig(registration);
+
+        ViewServiceConfig viewServiceConfig = null;
+
+        if (requestedViewServiceConfig instanceof IntegrationViewServiceConfig) {
+            /*
+             * The requested configuration is for an Integration View Service
+             */
+            IntegrationViewServiceConfig requestedIntegrationViewServiceConfig = (IntegrationViewServiceConfig)requestedViewServiceConfig;
+            IntegrationViewServiceConfig createdViewServiceConfig = new IntegrationViewServiceConfig(registration);
+            createdViewServiceConfig.setResourceEndpoints(requestedIntegrationViewServiceConfig.getResourceEndpoints());
+            viewServiceConfig = (ViewServiceConfig) createdViewServiceConfig;
+        }
+        else if (requestedViewServiceConfig instanceof SolutionViewServiceConfig) {
+            /*
+             * The requested configuration is for a Solution View Service
+             */
+            SolutionViewServiceConfig requestedSolutionViewServiceConfig = (SolutionViewServiceConfig)requestedViewServiceConfig;
+            SolutionViewServiceConfig createdViewServiceConfig = new SolutionViewServiceConfig(registration);
+            createdViewServiceConfig.setOMAGServerPlatformRootURL(requestedSolutionViewServiceConfig.getOMAGServerPlatformRootURL());
+            createdViewServiceConfig.setOMAGServerName(requestedSolutionViewServiceConfig.getOMAGServerName());
+            viewServiceConfig = (ViewServiceConfig) createdViewServiceConfig;
+        }
+        else {
+            /*
+             * Assume that the requested configuration is a vanilla view service configuration
```

Comment:
I suggest we fail the request if we do not recognise the view service as one of the 2 types we support

Reply:
We support all 3 types

Reply:
I thought we only support the integration type and the governance solution one, I am suggesting if it is not one of them we should fail the request as it is meaningless.